src/helpers.js

```diff
@@ -4,7 +4,7 @@ export const buildNoTypenameError = node =>
   )}`
 
 export const hasField = name => set =>
-  set.some(({ name: { value, }, }) => value === name)
+  set.some(({ alias, name }) => (alias || name).value === name)
```

Comment:
You are shadowing the `name` argument here so the comparison is not correct. Change the `name` argument to `fieldName` and fix the comparison.

Reply:
What? Dont understand what you mean. This is working code. Please provide your change you have in mind instead.

Reply:
Both `alias` and `name` are both objects if they exists else `undefined`.

## Scenario 1
```json
{
  "name": { "value": "something" }
}
```

## Scenario 2
```json
{
  "alias": { "value": "somethingElse" },
  "name": { "value": "something" }
}
```

## Scenario 3
```json
{
}
```

Reply:
The `hasField` argument - it's called `name`. You are shadowing it when destructuring with `{ alias, name }` which leads to self comparison (being always true, unless there is an alias).

do this:
```js
export const hasField = fieldName => set =>
  set.some(({ name: { value, }, }) => value === fieldName)
```

Reply:
actually, not "always `true`" because you are comparing against the `value` property, but still not correct

Reply:
Oh I didn’t see that. I normally have lint rules catching that.

Reply:
Hey, @EloB! When do you think this PR will be good to merge?

Reply:
I would recommend you of using this lint rule.
https://eslint.org/docs/rules/no-shadow